OpenAPI/LearningHub.Nhs.OpenApi/Controllers/OpenApiControllerBase.cs

```diff
@@ -0,0 +1,41 @@
+﻿namespace LearningHub.NHS.OpenAPI.Controllers
+{
+    using System.Net;
+    using System.Security.Claims;
+    using LearningHub.Nhs.OpenApi.Models.Exceptions;
+    using Microsoft.AspNetCore.Mvc;
+
+    /// <summary>
+    /// The base class for API controllers.
+    /// </summary>
+    public abstract class OpenApiControllerBase : ControllerBase
+    {
+        /// <summary>
+        /// Gets the current user's ID.
+        /// </summary>
+        public int? CurrentUserId
+        {
+            get
+            {
+                if ((this.User?.Identity?.AuthenticationType ?? null) == "AuthenticationTypes.Federation")
```

Comment:
> can you please let me know why we have a hard code value for authentication type?

Because we only have userId in this authentificationtype